File: carddemo-java/src/main/java/com/aws/carddemo/model/TransactionType.java
package com.aws.carddemo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * Transaction Type entity - migrated from COBOL copybook CVTRA03Y.cpy
 * Original COBOL record length: 60 bytes
 * Reference data for transaction types
 */
@Entity
@Table(name = "TRANTYPE")
public class TransactionType {

    @Id
    @Column(name = "TRAN_TYPE", length = 2)
    private String tranType;

    @Column(name = "TRAN_TYPE_DESC", length = 50)
    private String tranTypeDesc;

    public TransactionType() {
    }

    public TransactionType(String tranType) {
        this.tranType = tranType;
    }

    public String getTranType() {
        return tranType;
    }

    public void setTranType(String tranType) {
        this.tranType = tranType;
    }

    public String getTranTypeDesc() {
        return tranTypeDesc;
    }

    public void setTranTypeDesc(String tranTypeDesc) {
        this.tranTypeDesc = tranTypeDesc;
    }

    @Override
    public String toString() {
        return "TransactionType{" +
                "tranType='" + tranType + '\'' +
                ", tranTypeDesc='" + tranTypeDesc + '\'' +
                '}';
    }
}

File: carddemo-batch-python/src/carddemo_batch/models/transaction_type.py
"""TransactionType model - migrated from COBOL copybook CVTRA03Y.cpy (60 bytes)."""

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class TransactionType(Base):
    """
    Transaction type reference entity representing the TRANTYPE VSAM file.
    
    Migrated from COBOL copybook CVTRA03Y.cpy.
    Record length: 60 bytes
    Primary key: TRAN-TYPE (2 characters)
    """
    __tablename__ = "trantype"

    tran_type: Mapped[str] = mapped_column(String(2), primary_key=True)
    tran_type_desc: Mapped[str] = mapped_column(String(50), nullable=True)

    def __repr__(self) -> str:
        return f"TransactionType(type={self.tran_type}, desc={self.tran_type_desc})"
